src/wallet/wallet.cpp

```diff
@@ -1748,6 +1748,21 @@ int64_t CWallet::RescanFromTime(int64_t startTime, const WalletRescanReserver& r
     return startTime;
 }
 
+GCSFilter::ElementSet CWallet::GetScriptPubKeysForBlockFilter()
+{
+    GCSFilter::ElementSet ret;
+    for (const auto& i : CBasicKeyStore::GetAllWatchedScriptPubKeys()) {
+        ret.emplace(i.begin(), i.end());
```

Comment:
This will miss P2PKH outputs I think.

Reply:
Could you please elaborate in which code path we import a P2PKH address or a public key without adding it to `setWatchOnly`?

Reply:
I was wrong. Fixed.